shanoir-ng-front/src/app/users/user/user.component.ts

```diff
@@ -73,7 +76,15 @@ export class UserComponent extends EntityComponent<User> {
                 this.user.expirationDate = user.extensionRequestInfo.extensionDate;
             }
         });
-        Promise.all([userPromise, this.getRoles()]).then(() => {
+        let getMembershipPromise: Promise<void> = this.studyService.findStudiesByUserId().then(studies => {
+            this.studies = studies.filter(study =>  {
+                for( var suser of study.studyUserList) {
+                    // Admin case, we check that the user is part of the study
+                    return suser.userId === this.entity.id;
+                }
+            });
+        });
+        Promise.all([userPromise, getMembershipPromise, this.getRoles()]).then(() => {
```

Comment:
I don't think you had to add getMembershipPromise to this Promise.all .
You don't need to wait for the list of studies to set the role.

Reply:
Yep! I corrected it :)